crawler_message_item.go

```diff
@@ -34,6 +34,19 @@ func (c *CrawlerMessageItem) URL() string {
 	return string(c.Body)
 }
 
+func (c *CrawlerMessageItem) hasParams() bool {
+	urlParts, err := url.Parse(c.URL())
+	if err != nil {
```

Comment:
You could shorten this to just a single return statement:

```go
return (err != nil || urlParts.RawQuery != "")
```

Reply:
good spot